docs/src/docs/user-guide.adoc

```diff
@@ -0,0 +1,694 @@
+    private int id;
+    private String name;
+    private String organization;
+    private String position;
+
+    ...
+}
+
+public class CustomerAdapter implements JsonbAdapter<Customer, JsonObject> {
+    @Override
+    public JsonObject adaptToJson(Customer c) throws Exception {
+        return Json.createObjectBuilder()
+            .add("id", c.getId())
+            .add("name", c.getName())
+            .build();
+    }
+
+    @Override
+    public Customer adaptFromJson(JsonObject adapted) throws Exception {
+        Customer c = new Customer();
+        c.setId(adapted.getInt("id"));
+        c.setName(adapted.getString("name"));
+        return c;
+    }
+}
+----
+
+First scenario:
+
+[source,java]
+----
+// Create Jsonb with default configuration
+Jsonb jsonb = JsonbBuilder.create();
+
+// Create customer
+Customer c = new Customer();
+
+// Initialization code is skipped
+
+// Serialize
+jsonb.toJson(customer);
+----
+
+Result:
+
+[source,json]
+----
+{
+    "id": 1,
+    "name": "Json Bourne",
+    "organization": "Super Agents",
+    "position": "Super Agent"
+}
+----
+
+Second scenario:
+
+[source,java]
+----
+// Create custom configuration
+JsonbConfig config = new JsonbConfig()
+    .withAdapters(new CustomerAdapter());
+
+// Create Jsonb with custom configuration
+Jsonb jsonb = JsonbBuilder.create(config);
+
+// Create customer
+Customer c = new Customer();
+
+// Initialization code is skipped
+
+// Serialize
+jsonb.toJson(customer);
+----
+
+Result:
+
+[source,json]
+----
+{
+    "id": 1,
+    "name": "Json Bourne"
+}
+----
+
+
+==== Serializers/Deserializers
+
+Sometimes adapters mechanism is not enough and low level access to JSONP parser/generator is needed.
+
+From the spec:
+
+Serializer is a class implementing javax.json.bind.serializers.JsonbSerializer interface. It is used to serialize the type it’s registered on (Original). On serializing of Original type JSONB calls JsonbSerializer::serialize method. This method has to contain a custom code to serialize Original type using provided JsonpGenerator.
+
+Deserializer is a class implementing javax.json.bind.serializers.JsonbDeserializer interface. It is used to deserialize the type it’s registered on (Original). On deserialization of Original type JSONB calls JsonbDeserializer::deserialize method. This method has to contain a custom code to deserialize Original type using provided JsonpParser.
+
+Let's take a look at the sample. Imagine that we would like to serialize and deserialize a list of pet carriers. A carried pet defined by an abstract class Animal. It can be any of its subclasses. We would like to properly serialize and deserialize it. In order to do it we need to preserve a type information in JSON document and use it on deserialization. It can be done using custom serializer/deserializer pair.
+
+
+Model:
+
+[source,java]
+----
+public abstract class Animal {
+    private String name;
+    private int age;
+    private Boolean furry;
+
+    ...
```

Comment:
lets be more clear about what we mean by `. . .` in these classes. I assume public getters and setters (since the fields are private), but lets make that clear since the reader may not be able to infer what we mean